ADR/xxxx-kite-poc-architecture-and-components.md

```diff
@@ -12,6 +12,121 @@ The [Konflux Issue Tracking Engine](https://github.com/konflux-ci/kite) (KITE) i
+  - Each issue has logs to those builds, allowing for deeper investigation.
+- A power-user associate can use the [KITE CLI tool](https://github.com/konflux-ci/kite/tree/main/packages/cli) to quickly check if any existing issue records exist in the namespace(s) the user has access to.
+  - The CLI tool reports three build failures in namespace `team-gamma`, with logs for each failure.
+
+**Value**:
+- Non-technical support associates can leverage the issues dashboard to quickly show them what is happening where and why.
+- Power-user support associates can quickly find the information they need using the CLI tool, getting them closer to where deeper analysis is needed.
+- Regular users also have access to these tools, potentially reducing the amount of support tickets filed.
+
+**4. Security Vulnerability Response**
+
+**Scenario**: A critical CVE is discovered in a widely-used dependency (e.g. a container base image, Python package, or RPM package) that affects multiple components across different teams.
+
+**Current Problem**:
+- Teams are unaware that MintMaker's automated security updates are failing.
+- MintMaker runs Renovate to automatically create PRs/MRs with `[SECURITY]` tags for CVE fixes, but when these fail due to authentication issues, dependency conflicts, or breaking changes, there's no centralized visibility.
+- Teams only discover the security exposure during manual audits, security scans, or after incidents occur.
+
+**With Issues Dashboard**:
+```
+- Issue: "[SECURITY] Critical CVE-2024-1234 in base image - MintMaker updates failing"
+- Severity: Critical
+- Affected Scope: 15 components across 6 namespaces
+- Failure Details:
+  • 4 components: GitLab Renovate token expired in namespace secrets
+  • 7 components: Container registry authentication failed
+  • 3 components: Dependency version conflicts preventing clean updates
+  • 1 component: Repository lacks required GitHub App installation
+- Direct Links:
+  • Failed Renovate execution logs for each component
+  • CVE vulnerability details and severity assessment
+  • Registry credential setup instructions
+  • Dependency conflict resolution guide
+```
+
+**Resolution Workflow**:
+1. **Immediate Alert**: All affected teams see the security issue in their dashboard when MintMaker detects failures
+2. **Targeted Remediation**: Teams can address specific failure causes (refresh tokens, fix registry auth, resolve conflicts)
+3. **Progress Tracking**: As MintMaker successfully creates security PRs for each component and those PRs get merged, the issue scope automatically shrinks
+4. **Auto-Resolution**: Issue resolves completely when all components have successful security updates merged
+
+**Key Value**:
+- Instead of X teams independently discovering a critical security vulnerability days or weeks later through security scanners, all teams are immediately aware with specific remediation steps.
```

Comment:
Today konflux allows to upload the sbom of a component to an SBOM server (for example RHTPA) that will show the cves related to the artifact. Isn't it enough?

Reply:
I don't think we intend to replicate the features of an SBOM analysis service; we have data from both minitmaker and Clair which are not surfaced to users in such a way that it is easily actionable and this is an example of how to surface that data better.